xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -70,6 +91,12 @@ void CProcessorHD::Close()
   m_pVideoProcessor = nullptr;
   m_pVideoContext = nullptr;
   m_pVideoDevice = nullptr;
+
+  // restores 10 bit swap chain if previously forced to 8 bit
+  if (m_forced8bit)
+    DX::DeviceResources::Get()->ApplyDisplaySettings();
```

Comment:
Is it necessary to add another mechanism to change/restore the swap chain? It's mostly taken care of automatically already by the renderer (change added not long ago for the AMD colorspace fix)

I think the only piece missing for SDR was to set the initial hdr type m_HdrType to a new value HDR_TYPE::HDR_UNKNOWN, so that CRendererBase::ProcessHDR performs the colorspace change/swap chain creation when starting playback of SDR.

Reply:
I don't quite understand what you mean... this has nothing to do with HDR but when swap chain is in 10 bit SDR and method is not new, already used here:

https://github.com/xbmc/xbmc/blob/31124ad11bd289a115851244ac953b8a181b0a89/xbmc/settings/DisplaySettings.cpp#L330

Reply:
CRendererBase::ProcessHDR is called unconditionally for both SDR and HDR.
it adjusts HDR on/off, color space, metadata, swap chain.
So it seemed like a natural place to extend instead of adding another logic.

But actually as written in other comment, I think it's the intermediate surface that should change format and the swap chain should be untouched.

Reply:
> But actually as written in other comment, I think it's the intermediate surface that should change format and the swap chain should be untouched.

Not relevant for this PR.